### ISSUE TITLE
CUDA vs CPU coordinate filtering inconsistency in PointToVoxel

### ISSUE THREAD

**Issue by akozlov-outrider** [traveller59/spconv#755]
# CUDA vs CPU coordinate filtering inconsistency in PointToVoxel

## Issue Description

There's an inconsistency between CPU and CUDA implementations of `PointToVoxel.generate_voxel_with_id()` where the CUDA version filters out more points than the CPU version, despite using identical parameters and input data.

## Environment
- SpConv version: [Please specify your version]
- PyTorch version: [Please specify your version] 
- CUDA version: [Please specify your version]
- Python version: [Please specify your version]

## Expected Behavior
Both CPU and CUDA implementations should filter the same number of points when given identical:
- Input point cloud data
- Voxel size (`vsize_xyz`)
- Coordinate range (`coors_range_xyz`)
- Other parameters

## Actual Behavior
CUDA implementation filters out more points than CPU implementation. In the reproduction case below, CUDA consistently filters out additional points compared to CPU.

## Reproduction Code

```python
import numpy as np
import torch
from spconv.pytorch.utils import PointToVoxel

# Create test point cloud with coordinates within bounds
def create_bounded_point_cloud(num_points: int, num_features: int = 4):
    # Generate coordinates within bounds: xy in [-52, 52], z in [-3, 8]
    coords = np.random.uniform(
        low=[-52, -52, -3],
        high=[52, 52, 8],
        size=[num_points, 3]
    ).astype(np.float32)
    
    # Generate additional features
    if num_features > 3:
        extra_features = np.random.randn(num_points, num_features - 3).astype(np.float32)
        features = np.concatenate([coords, extra_features], axis=1)
    else:
        features = coords
    
    # Add batch dimension
    batch_idx = np.zeros((num_points, 1), dtype=np.float32)
    elements = np.concatenate([batch_idx, features], axis=1)
    
    return torch.from_numpy(elements)

# Set random seed for reproducibility
np.random.seed(42)
point_features = create_bounded_point_cloud(1000, num_features=4)

# Test CPU
point_to_voxel_cpu = PointToVoxel(
    vsize_xyz=[0.2, 0.2, 0.2],
    coors_range_xyz=[-60, -60, -5, 60, 60, 10],
    num_point_features=4,
    max_num_voxels=50000,
    max_num_points_per_voxel=5,
    device=torch.device('cpu'),
)

_, _, _, point_to_voxel_map_cpu = point_to_voxel_cpu.generate_voxel_with_id(
    point_features, empty_mean=False
)
valid_cpu = (point_to_voxel_map_cpu >= 0).sum()

# Test CUDA
point_features_cuda = point_features.to('cuda')
point_to_voxel_cuda = PointToVoxel(
    vsize_xyz=[0.2, 0.2, 0.2],
    coors_range_xyz=[-60, -60, -5, 60, 60, 10],
    num_point_features=4,
    max_num_voxels=50000,
    max_num_points_per_voxel=5,
    device=torch.device('cuda'),
)

_, _, _, point_to_voxel_map_cuda = point_to_voxel_cuda.generate_voxel_with_id(
    point_features_cuda, empty_mean=False
)
valid_cuda = (point_to_voxel_map_cuda >= 0).sum()

print(f'Valid points on CPU: {valid_cpu}')
print(f'Valid points on CUDA: {valid_cuda}')
print(f'Difference: {valid_cpu - valid_cuda} more points filtered by CUDA')
```

## Additional Notes

- All input points are generated within the coordinate bounds [-52, 52, -52, 52, -3, 8] which is well within the specified `coors_range_xyz` of [-60, -60, -5, 60, 60, 10]
- The issue is reproducible with the same random seed
- This inconsistency could lead to different model behaviors when switching between CPU and CUDA inference

## Impact

This inconsistency can cause:
1. Different model outputs when switching between CPU and CUDA
2. Unreliable model behavior in production environments
3. Difficulty in debugging and validating model performance

Would appreciate any insights into why this difference occurs and if there's a fix or workaround available.
